modules/renter/contractor/update.go

```diff
@@ -0,0 +1,33 @@
+package contractor
+
+import (
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// renewThreshold calculates an appropriate renewThreshold for a given period.
+func renewThreshold(period types.BlockHeight) types.BlockHeight {
+	threshold := period / 4
+	if threshold > 1000 {
+		threshold = 1000
+	}
+	return threshold
+}
```

Comment:
also worth noting, because of the new support for virtual sectors, there's not a huge cost to renewing at half or even 75%, the cost is in the number of transactions that hit the blockchain, and not because you're going to be double-paying. I'd recommend at this point that the threshold default to 6 weeks, and that the period default to 12 weeks.